app/controllers/search_controller.rb

```diff
@@ -34,6 +34,25 @@ def show
       raise Discourse::InvalidParameters.new("page parameter must not be greater than 10")
     end
 
+    if request.user_agent &&
+         CrawlerDetection.crawler?(request.user_agent, request.headers["HTTP_VIA"])
+      crawler_html = <<~HTML
+          <!DOCTYPE html>
+          <html>
+            <head>
+              <meta name='robots' content='noindex'>
+            </head>
+            <body>
+              <p><em>*waves hand*</em> This is not the content you are looking for</p>
+            </body>
+          </html>
+        HTML
+
+      response.headers["X-Robots-Tag"] = "noindex"
+
+      return(render html: crawler_html.html_safe, layout: false, content_type: "text/html")
+    end
```

Comment:
I think we should be able to reject this request even earlier in a `before_action` scoped to the `show` action.

Reply:
I put it where it currently is to ensure that the various `InvalidParameters` checks are done first, since we don't want anyway to get a 200 status if they make a request with invalid parameters.

Reply:
> since we don't want anyway to get a 200 status if they make a request with invalid parameters.

I see. The middleware implementation wouldn't have been able to do this then so implementing this at the controller level was probably right.